cmd/gpu-kubelet-plugin/device_state.go

```diff
@@ -320,6 +337,19 @@ func (s *DeviceState) prepareDevices(ctx context.Context, claim *resourceapi.Res
 		}
 	}
 
+	if featuregates.Enabled(featuregates.DeviceHealthCheck) {
+		klog.V(6).Infof("Adding devices health status to claim %s/%s", claim.Namespace, claim.Name)
+		if err := s.applyClaimDeviceStatuses(ctx, claim.Namespace, claim.Name, deviceStatuses...); err != nil {
+			klog.Warningf("Failed to update devices status for claim %s/%s: %v", claim.Namespace, claim.Name, err)
+		}
```

Comment:
High level question -- who is going to be reading this status from the ResourceClaim and doing anything with it? I know I suggested looking at DRANet and seeing how they were reporting their health status, but does it even make sense to do this in our driver? Why does DRANet need to do it?

/cc @aojea